examples/simulation/plot_stc_metrics.py

```diff
@@ -0,0 +1,262 @@
+"""
+===============================================
+Compare simulated and estimated source activity
+===============================================
+
+This example illustrates how to compare the simulated and estimated
+source time courses (STC) by computing different metrics. Simulated
+source is a cortical region or dipole. It is meant to be a brief
+introduction and only highlights the simplest use case.
+
+"""
+# Author: Kostiantyn Maksymenko <kostiantyn.maksymenko@gmail.com>
+#
+# License: BSD (3-clause)
+
+import os.path as op
+
+import numpy as np
+import matplotlib.pyplot as plt
+from functools import partial
+
+import mne
+from mne.datasets import sample
+from mne.minimum_norm import make_inverse_operator, apply_inverse
+from mne.simulation.metrics import (region_localization_error,
+                                    f1_score, precision_score,
+                                    recall_score, cosine_score,
+                                    peak_position_error,
+                                    spatial_deviation_error)
+
+print(__doc__)
+
+random_state = 42  # set random state to make this example deterministic
+
+# Import sample data
+data_path = sample.data_path()
+subjects_dir = op.join(data_path, 'subjects')
+subject = 'sample'
+evoked_fname = op.join(data_path, 'MEG', subject, 'sample_audvis-ave.fif')
+info = mne.io.read_info(evoked_fname)
+tstep = 1. / info['sfreq']
+
+# Import forward operator and source space
+fwd_fname = op.join(data_path, 'MEG', subject,
+                    'sample_audvis-meg-eeg-oct-6-fwd.fif')
+fwd = mne.read_forward_solution(fwd_fname)
+src = fwd['src']
+
+# To select source, we use the caudal middle frontal to grow
+# a region of interest.
+selected_label = mne.read_labels_from_annot(
+    subject, regexp='caudalmiddlefrontal-lh', subjects_dir=subjects_dir)[0]
+
+
+###############################################################################
+# In this example we simulate two types of cortical sources: a region and
+# a dipole sources. We will test corresponding performance metrics.
+
+###############################################################################
+# Define main parameters of sources
+# ---------------------------------
+#
+# First we define both region and dipole sources in terms of
+# Where?, What? and When?.
+
+# WHERE?
```

Comment:
I don't fully understand the combination of `label`, `location` and `extent` parameters in `select_sources`.